include/cuda/std/detail/libcxx/include/optional

```diff
@@ -236,8 +283,13 @@ struct __optional_destruct_base<_Tp, false>
         :  __val_(_CUDA_VSTD::forward<_Args>(__args)...),
            __engaged_(true) {}
 
+  template <class _Fp, class... _Args>
+  _LIBCUDACXX_INLINE_VISIBILITY
+  constexpr __optional_destruct_base(__optional_construct_from_invoke_tag, _Fp&& __f, _Args&&... __args)
+      : __val_(_CUDA_VSTD::invoke(_CUDA_VSTD::forward<_Fp>(__f), _CUDA_VSTD::forward<_Args>(__args)...)), __engaged_(true) {}
+
     _LIBCUDACXX_INLINE_VISIBILITY
-    void reset() noexcept
+    constexpr void reset() noexcept
```

Comment:
This will work at constexpr in pre-C++17 if the branch isn't engaged though, right? Though maybe that's not useful enough to make it unconditionally constexpr?

Reply:
I'll make it conditional.